### PR TITLE
Fix reading `nonlinear` input for GENE

### PR DESCRIPTION
**Pull request by ZedThree**
Was missing namelist name

### CODE REVIEW

**Comment by bpatel2107** [comment]
I added in the set up for `kx`, `ky` and the fluxes too

File: pyrokinetics/gk_code/GKOutputReaderGENE.py [hunk 1]
```diff
@@ -309,12 +320,12 @@ def _set_fluxes(
         """
         # TODO This was changed to include a ky coordinate to match GS2 and CGYRO.
         #     Should this be reverted?
```
Comment (ZedThree) [inline pyrokinetics/gk_code/GKOutputReaderGENE.py]: This comment can probably go now!

File: pyrokinetics/gk_code/GKInputGENE.py [hunk 2]
```diff
@@ -78,7 +78,7 @@ def write(self, filename: PathLike, float_format: str = ""):
         super().write(filename, float_format=float_format)
 
     def is_nonlinear(self) -> bool:
-        return bool(self.data.get("nonlinear", 0))
+        return bool(self.data["general"].get("nonlinear", 0))
```
Comment (ZedThree) [inline pyrokinetics/gk_code/GKInputGENE.py]:
```suggestion
        return self.data["general"].get("nonlinear", False)
```

Comment (bpatel2107) [reply pyrokinetics/gk_code/GKInputGENE.py]: Should this been a boolean or a float?

Comment (ZedThree) [reply pyrokinetics/gk_code/GKInputGENE.py]: Oh, I see you've changed the other bit to be like this -- does GENE set nonlinear to 1/0 rather than T/F?

Comment (bpatel2107) [reply pyrokinetics/gk_code/GKInputGENE.py]: GENE sets it to a boolean so what you have suggested is right

Comment (ZedThree) [reply pyrokinetics/gk_code/GKInputGENE.py]: It's supposed to return a bool. I only have one example GENE input file and it has `nonlinear = F`, but I forgot 1/0 are valid inputs for bools in Fortran. So I would have:

```suggestion
        return bool(self.data["general"].get("nonlinear", False))
```

Comment (bpatel2107) [reply pyrokinetics/gk_code/GKInputGENE.py]: Oh god I keep changing it. This seems best

**Comment by ZedThree** [comment]
The test is wrong :facepalm:

**Comment by bpatel2107** [comment]
LGTM

**Pull request merged**